doc/smart-switch/reboot/reboot-hld.md

```diff
@@ -0,0 +1,379 @@
+
+#### PCIE_DETACH_INFO schema in StateDB
+
+```
+"PCIE_DETACH_INFO|DPU_0": {
+  "value": {
+    "dpu_id": "0",
+    "dpu_state": "detaching",
+    "bus_info" : "[DDDD:]BB:SS.F"
+  }
+}
+```
+
+### PCIe daemon modifications ###
+The PCIe daemon will be updated to avoid logging "PCIe Device: <Device name> Not Found" messages when DPUs are undergoing a reboot, as this is a
+user-initiated action.
+
+In the [check_pci_devices()](#https://github.com/sonic-net/sonic-platform-daemons/blob/bf865c6b711833347d3c57e9d84cd366bcd1b776/sonic-pcied/scripts/pcied#L155) function,
+read the State DB for the PCIE_DETACH_INFO and suppress the "device not found" warning logs during a DPU reboot when the device is intentionally detached.
+
+### Error handling and exception scenarios ###
+
+The following are specific error scenarios where the DPU state will not be DPU_READY.
+
+* If the gNOI service is not operational on the DPU or DPU is unreachable for any reason, detach the PCI, and proceed with the reboot after a timeout
+upon receiving an acknowledgment.
+
+* After the DPU reboots, if the DPU PCI fails to reconnect for any reason, an error-handling mechanism should be in place to restore the DPU.
+
+* If a DPU fails to reboot during a switch reboot, the NPU should attempt to recover the DPU and log any errors that occur.
+
+* In the event of power failure, a power-cycle due to a kernel panic, or any other unknown reason, both the DPU and NPU will undergo an ungraceful reboot.
+
+* In the event of a DPU reboot failure, a hardware watchdog is needed to monitor and reset the DPU. This implementation is vendor-specific.
+
+## Test plan ##
+
+Presented below is the test plan within the ```sonic-mgmt``` framework for the smart switch reboot.
+
+
+| Event                                     | NPU reboot sequence | DPU reboot sequence |
+| ----------------------------------------- | ------------------- | ------------------- |
+| Power-On                                  | Graceful boot       | Graceful boot       |
+| Planned cold reboot of Smart Switch       | Graceful reboot     | Graceful reboot     |
+| Planned cold reboot of DPU                | -                   | Graceful reboot     |
+| Planned power-cycle of Smart Switch       | Graceful reboot     | Graceful reboot     |
+| Planned power-cycle of DPU                | -                   | Graceful reboot     |
+| Unplanned DPU power failure               | -                   | Ungraceful reboot   |
+| Unplanned Smart Switch power failure      | Ungraceful reboot   | Ungraceful reboot   |
+| Unplanned Smart Switch System Crash       | Ungraceful reboot   | Ungraceful reboot   |
+| Unplanned DPU System Crash                | -                   | Ungraceful reboot   |
+
+The test scenarios above ensure that both the NPU and all DPUs are fully operational following any type of reboot. Furthermore, the tests verify the
+functionality of PCI communication between NPU and DPUs.
+
+## References
```

Comment:
@vvolam please put reference to PMON HLD design for smart switch

Reply:
Updated.